### PR TITLE
`camera-target-updated` (was `center-changed`) event 

### PR DESCRIPTION
**Pull request by autra**
This PR correctly set and fire a `camera-target-updated` when the target of the camera on the Globe changes. This event was called `center-changed`, but arguably the new name is clearer.

This is WIP because it's based on peep's work in #239. As #239 is a core work, let's try to merge it quickly please!

Notifying @pjjmunier @elias75015 and @gcebelieu this is the first PR fixing the events you need, according to [this list](https://github.com/iTowns/itowns2/issues/272#issuecomment-287779100). Don't hesitate to test!

Notifying also @vcoindet : I believe this PR overlaps a bit was you did in #269, but it is only about one event, and pushed things a bit farther up (notably: it does not only fire when explicitly calling the API, but also when user interacts with the mouse, which is the intended behaviour I believe).

### CODE REVIEW

**Comment by autra** [comment]
Btw, the event is available on the current controls (currently available at `itowns.viewer.scene.currentControls()`,  though I hope we would simplify that in the future). Amending the doc to reflect that.

**Comment by gchoqueux** [comment]
> I believe this PR overlaps a bit was you did in #269 

It may have been necessary to agree with @vcoindet before running. it's the case?

**Comment by autra** [comment]
> It may have been necessary to agree with @vcoindet before running. it's the case?

Oh I didn't see the last changes on @vcoindet's PR, my bad :-) Anyway, it was faster (and probably better) to show some code than explaining what I had in mind, because my PR also contains some renaming. Btw, don't worry, I don't care which PR gets merged, really :-) I do care about having something that works on master. So if you want, I can comment on @vcoindet's PR to help things move on.

**Comment by gchoqueux** [comment]
OK, We will mix the code, if necessary

**Comment by vcoindet** [comment]
OK I saw your code @autra, it seems to be exactly what I did in #269...
I don't care wich PR is merged, but I'd like to test your code first to be sure it does the same thing. ;-)

**Comment by autra** [comment]
We can get yours merged if you want to @vcoindet, I'm completely ok with that :-) Do you want me to comment there?

**Comment by elias75015** [comment]
@autra thanks for your PR. We will quickly test it, but it seems that a great part of your code is corresponding to what @vcoindet did on his PR #269.

As seen with Victor, he will mix the code of both proposals. Once this work done, we will test the event handling more deeply :)

**Comment by autra** [comment]
No problem :-) btw what do you guys think about the rename of center?

**Comment by elias75015** [comment]
We're okay with the rename to camera-target-updated, it makes sense :+1:  

The most important is that the event is correctly triggered and returns to the user info expected !

**Comment by vcoindet** [comment]
I still didn't had time to test, but I'd rather chose your PR @autra as you already rename the center. I will remove what I've done in mine concerning the center event.
I'm OK with the new names btw, we still understand what this is about ! ;-)

**Comment by autra** [comment]
Yeah I thought center was too generic...

okay, whatever you want guys :-) 

> As seen with Victor

For cases like that, it may be better to discuss  on the open (irc channels, github etc...) than in private @elias75015 . That's the way open-source projects do :-)

**Comment by vcoindet** [comment]
OK I tested it. Works fine for me.

I guess we can merge this PR, I will adapte my PR in consequences.

**Comment by vcoindet** [comment]
Can you please wait before merging this PR ?

We would like to discuss the point that `oldCameraTargetPosition` and `newCameraTargetPosition` are in cartesian and it seems a bit strange for a globe. I think we should be able to propose those values in degrees for this case.

**Comment by gcebelieu** [comment]
Shouldn't it be in the current coordinates system of the map / globe along with the CRS id in wich it is returned ?

**Comment by elias75015** [comment]
Indeed, I think the most common behaviour is to return values in the coordinates system of the map .

**Comment by autra** [comment]
The logic here is to return coordinates from Three.js (which happens to be cartesian in this use-case, but might not in other examples).  The idea is also to support multiple data-sources each having its own CRS. In this case, the only common denominator is to return Three.js coordinates, and let the user convert according to his needs. Imho, it's the simplest yet most flexible solution. As [shown in `Debug.js`](https://github.com/iTowns/itowns2/pull/277/files#diff-dd325aead9f06453147e0a0a07638301R239), the conversion is super-easy anyway.

Anyway, after having thought about that, I even think we shouldn't give back any informations in our events as a first iteration of this. We don't really know which informations are relevant for the client anyway. Some client code might just want to be notified in some case, maybe some wants only the new position etc. 

So maybe the right first step would be to remove all info and let client code query `getCameraTargetPosition` if they really want coordinates in their event handler. Later, if we realize that all or most listeners also wants some other infos, we could add them. Thoughts?

**Comment by autra** [comment]
Also, embedding nothing might be better for performance for events that are fired very often (during moves for instance), though I don't know yet if this performance hit is significant.

**Comment by autra** [comment]
Actually, I think I can embed 2 Coordinates object in the event, It may be the best thing to do at the end and will allow conversion between different CRS.

Btw, yes, this PR is not ready to get merged, simply because it depends on another one (#239)

**Comment by elias75015** [comment]
I think it is not really a problem for us if the coordinates are not directly in the CRS of the globe (geographic), if you think it is better that way. 
In theory, we can convert them pretty easily by using the "_convert" or the "cartesianToGeo"  functions (see Coordinates.js and Projection.js). We still need to test these functions.

Anyway, we should at least have coordinates informations embed in the event.

**Comment by autra** [comment]
> In theory, we can convert them pretty easily by using the "_convert"

You mean `Coordinates.as`.

> Anyway, we should at least have coordinates informations embed in the event.

Well, it's not strictly necessary (because you can query them through methods in the api), but I'll embed Coordinates objects anyway.

Btw, Coordinates objects are the "right" way of using these informations in Itowns. Under the hood, they delegate to proj4 for the conversions.

**Comment by gchoqueux** [comment]
> The logic here is to return coordinates from Three.js

In Ellipsoid, the logic isn't the cartesian. In itowns, ellipsoid building is based on spherical coordinate system, we build our ellipsoid in a spherical system and we convert it to cartesian for THREE.js. In this case,  WGS84 is spherical coordinate system. 

This system is not practical and will always be converted to place objects or make measurements.
The data returned must be the most suitable for the system in which we work.

**Comment by autra** [comment]
> The data returned must be the most suitable for the system in which we work.

What I've said earlier is not relevant any more anyway, but please note we don't always have ellipsoid, and we don't always work in WGS84.

**Comment by gchoqueux** [comment]
Yes, quite, but an adapted coordinate system must always be the case

**Comment by autra** [comment]
> an adapted coordinate system must always be the case

Your sentence is a bit unclear. I *think* we agree, but can you rephrase please?

**Comment by autra** [comment]
Rebased on master.

File: src/Core/Geographic/Coordinates.js
Comment (gchoqueux) [inline src/Core/Geographic/Coordinates.js]: * You have reversed longitude and latitude
* `cartesianToCartographic` return in Radian and EPSG:4326 is in degree

File: src/Renderer/ThreeExtended/GlobeControls.js
Comment (gchoqueux) [inline src/Renderer/ThreeExtended/GlobeControls.js]: typo : target

File: src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js
Comment (gchoqueux) [inline src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js]: Warning, bad WGS84 axes, see #246, Y is up axis in iTowns and WGS84 Z is the up axis
Maybe add a comment to wait fix in #246
As you pay attention to the order of the axes?

File: utils/debug/Debug.js
Comment (gchoqueux) [inline utils/debug/Debug.js]: Displays less decimal for readability

**Review by gchoqueux — CHANGES_REQUESTED** [review]

File: src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js
Comment (autra) [reply src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js]: Not sure how it is related to my change. I only added a parameter that was hardcoded before. I don't really deal with axes there.

**Comment by autra** [comment]
updated

File: src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js
Comment (gchoqueux) [reply src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js]: Yes, but it was not explicitly stated that the scene respected the 'EPSG:4978', because precisely it was not respected. So, it doesn't hurt to leave a little warning

Comment (autra) [reply src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js]: In this case, I need to think about it. Maybe then `EPSG:4978` is simply and plainly incorrect and I need to wait for #246 to get merged for this to make any sense. I won't but a `// warning: this is wrong` in the code :-) 

At any rate, no time for this now, I'll see next week.

File: src/Core/Geographic/Coordinates.js
Comment (gchoqueux) [inline src/Core/Geographic/Coordinates.js:115]: `cartesianToCartographic` returns in radian and `'EPSG:4326'` is in DEGREE

File: src/Core/Geographic/Projection.js
Comment (gchoqueux) [inline src/Core/Geographic/Projection.js:173]: You must inverse the axis
In debug console, the coordinates radian is wrong

**Review by gchoqueux — CHANGES_REQUESTED** [review]

**Comment by autra** [comment]
@gchoqueux updated.

**Review by gchoqueux — APPROVED** [review]

**Pull request merged**
